src/webgpu/util/math.ts

```diff
@@ -317,3 +320,9 @@ export function multiplyMatrices(
 
   return product;
 }
+
+/** Sign-extend the `bits`-bit number `n` to a 32-bit signed integer. */
+export function signExtend(n: number, bits: number): number {
```

Comment:
Why we need this?

Reply:
This is used in the next PR, but I included it here since it's easy to review on its own.

Reply:
Ack.